torch/optim/optimizer.py

```diff
@@ -789,6 +789,7 @@ def zero_grad(self, set_to_none: bool = True) -> None:
                 (in one case it does the step with a gradient of 0 and in the other it skips
                 the step altogether).
         """
+        from torch._dynamo.utils import get_static_address_type
```

Comment:
Should this be within the zero_grad function or outside?

Also, there are other places people can call zero_grad (e.g., module.zero_grad). Just FYI.

Reply:
There is a circular dependency issue with importing torch._dynamo.utils in the optimizer module, this is why all the dynamo imports in this file are local. It's actually non-trivial to resolve as well so we usually duplicate APIs to torch.utils. This api seemed pretty dynamo specific, and I'm only using this in one place so I didn't think it was worth it in this case.

Reply:
Is it in any roadmap to resolve the circular dependency?

Reply:
You could just use `torch._dynamo.utils.get_static_address_type()` directly below to remove this import